pkg/controller/v1alpha2/applicationconfiguration/applicationconfiguration.go

```diff
@@ -228,6 +228,7 @@ func (r *OAMApplicationReconciler) Reconcile(req reconcile.Request) (result reco
 
 	// execute the posthooks at the end no matter what
 	defer func() {
+		updateObservedGeneration(ac)
```

Comment:
An afterthought: I don't think this is the right place to update ObservedGeneration. Because this will always update even if the reconciliation failed.

Note that I will be fixing this shortly.

@wonderflow @Ghostbaby

Reply:
Reconcile failed also need to add observed